zilliqa/src/api/types/zil.rs

```diff
@@ -102,41 +110,50 @@ pub struct CreateTransactionResponse {
 
 #[derive(Clone, Serialize, Debug)]
 struct GetTxResponseReceipt {
-    cumulative_gas: String,
-    epoch_num: String,
+    #[serde(with = "num_as_str")]
+    cumulative_gas: u64,
+    #[serde(with = "num_as_str")]
+    epoch_num: u64,
     success: bool,
     #[serde(skip_serializing_if = "Option::is_none")]
-    event_logs: Option<Vec<Value>>,
+    event_logs: Option<Vec<ScillaLog>>,
 }
 
 impl GetTxResponse {
-    pub fn new(verified_tx: VerifiedTransaction, receipt: TransactionReceipt) -> Option<Self> {
-        match verified_tx.tx {
-            // todo: make all of the fields correct
-            SignedTransaction::Zilliqa { ref tx, .. } => Some(GetTxResponse {
-                id: verified_tx.hash.to_string(),
-                version: "65537".to_string(),
-                nonce: tx.nonce.to_string(),
-                to_addr: tx.to_addr, // Note this appears to have no 0x prefix in zq1
-                sender_pub_key: hex::encode(verified_tx.signer),
-                amount: tx.amount.to_string(),
-                signature: format!(
-                    "0x{}{}",
-                    hex::encode(verified_tx.tx.sig_r()),
-                    hex::encode(verified_tx.tx.sig_s())
-                ),
+    pub fn new(
+        tx: VerifiedTransaction,
+        receipt: TransactionReceipt,
+        block_number: u64,
+    ) -> Option<Self> {
+        let VerifiedTransaction { tx, hash, .. } = tx;
+        if let SignedTransaction::Zilliqa { tx, key, sig } = tx {
+            Some(GetTxResponse {
+                id: H256(hash.0),
+                version: ((tx.chain_id as u32) << 16) | 1,
+                nonce: tx.nonce,
+                to_addr: tx.to_addr,
+                sender_pub_key: key,
+                amount: tx.amount,
+                signature: sig,
                 receipt: GetTxResponseReceipt {
-                    cumulative_gas: receipt.gas_used.to_string(),
-                    epoch_num: "1".to_string(), // todo here
+                    cumulative_gas: receipt.gas_used,
+                    epoch_num: block_number,
```

Comment:
Do you mean this?

Reply:
Yes - In this response `epoch_num` means "TX block in which this transaction was mined" (or at least that's how ZQ1 behaves).

I agree the name is bad, especially since 'epoch' has its own meaning in ZQ2.

Reply:
Ah! Marking resolved, but it might be worth a comment in the code, else people will keep saying this to you until you do :-)